components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java

```diff
@@ -250,44 +271,51 @@ public static void validateRuleTemplateProperty(RuleTemplateProperty ruleTemplat
+        Collection<Template> templates = ruleTemplate.getTemplates();
+        Map<String, RuleTemplateProperty> properties = ruleTemplate.getProperties();
         Collection<String> templatedElements = new ArrayList();
 
-        // Add all templated elements to Collection
-        for (Template template : templates) {
-            String templatedContent = template.getContent();
+        if (ruleTemplate.getScript().isEmpty()) {
+            // Add all templated elements to Collection
+            for (Template template : templates) {
 
-            // Find all templated elements from the siddhiApp
-            Pattern templatedElementPattern = Pattern.compile(TemplateManagerConstants.TEMPLATED_ELEMENT_REGEX_PATTERN);
-            Matcher templatedElementMatcher = templatedElementPattern.matcher(templatedContent);
+                String templatedContent = template.getContent();
 
-            // When each templated element is found
-            while (templatedElementMatcher.find()) {
-                // Add templated element (inclusive of template pattern)
-                String templatedElement = templatedElementMatcher.group(1);
+                // Find all templated elements from the siddhiApp
+                Pattern templatedElementPattern = Pattern.compile(TemplateManagerConstants.TEMPLATED_ELEMENT_REGEX_PATTERN);
+                Matcher templatedElementMatcher = templatedElementPattern.matcher(templatedContent);
 
-                // Find Templated Element's Name
-                Pattern templatedElementNamePattern = Pattern.compile(TemplateManagerConstants.TEMPLATED_ELEMENT_NAME_REGEX_PATTERN);
-                Matcher templatedElementNameMatcher = templatedElementNamePattern.matcher(templatedElement);
+                // When each templated element is found
+                while (templatedElementMatcher.find()) {
+                    // Add templated element (inclusive of template pattern)
+                    String templatedElement = templatedElementMatcher.group(1);
 
-                // When the Templated Element's Name is found
-                if (templatedElementNameMatcher.find()) {
-                    // Templated Element's Name
-                    String templatedElementName = templatedElementNameMatcher.group(1);
+                    // Find Templated Element's Name
+                    Pattern templatedElementNamePattern = Pattern.compile(TemplateManagerConstants.TEMPLATED_ELEMENT_NAME_REGEX_PATTERN);
+                    Matcher templatedElementNameMatcher = templatedElementNamePattern.matcher(templatedElement);
 
-                    templatedElements.add(templatedElementName);
+                    // When the Templated Element's Name is found
+                    if (templatedElementNameMatcher.find()) {
+                        // Templated Element's Name
+                        String templatedElementName = templatedElementNameMatcher.group(1);
+
```

Comment:
unnecessary new line

Reply:
This method was removed by 3674cdddd78974cf7b8aca6a5feb11440c7c499b